contracts/RewardsManager.vy

```diff
@@ -3,18 +3,46 @@
 # @license MIT
 from vyper.interfaces import ERC20
 
+struct TokenReward:
+    gift_token: address
+    scale: uint256
+    duration: uint256
+    reward_distribution: address
+    period_finish: uint256
+    reward_rate: uint256
+    last_update_time: uint256
+    reward_per_token_stored: uint256
+
 interface FarmingRewards:
-    def tokenRewards(index: uint256) -> (address, uint256, uint256, address, uint256, uint256, uint256, uint256): view
+    def tokenRewards(index: uint256) -> TokenReward: view
     def notifyRewardAmount(index: uint256, reward: uint256): nonpayable
 
+
+event OwnershipTransferred: 
+    previous_owner: indexed(address)
+    new_owner: indexed(address)
+
+event RewardsContractSet:
+    rewards_contract: indexed(address)
+
+event ERC20TokenRecovered:
+    token: indexed(address)
+    amount: uint256
+    recipient: indexed(address)
+
 owner: public(address)
-gift_index: public(uint256)
+GIFT_INDEX: constant(uint256) = 1
 rewards_contract: public(address)
 ldo_token: constant(address) = 0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32
 
 @external
-def __init__():
+def __init__(_rewards_contract: address):
+    assert _rewards_contract != ZERO_ADDRESS, "zero address"
+    self.rewards_contract = _rewards_contract
+
     self.owner = msg.sender
+    log OwnershipTransferred(ZERO_ADDRESS, msg.sender)
```

Comment:
I guess we miss another event here:
```vyper
log RewardsContractSet(_rewards_contract)
```
Otherwise, do we really use this event anywhere?